token-metadata/program/src/processor.rs

```diff
@@ -1869,7 +1872,7 @@ pub fn set_collection_size(
             }
         }
     } else {
-        return Err(MetadataError::NotACollectionParent.into());
+        metadata.collection_details = Some(CollectionDetails::V1 { size });
```

Comment:
> > This branch can be replaced with just:  ...
> 
> I'm not saying that the code should be changed, I'm talking about a unit test that ensures that the code behaves as expected since there is new logic added to the code.

I'll add a test.

Reply:
> Would it make sense to add a log message here just to note that collection details are being added that were not previously here?

I think this isn't necessary anymore with the refactor in the latest commit. If you disagree, lmk and I'll add a log message.